server/app.js

```diff
@@ -1,34 +1,48 @@
-/* eslint-disable import/no-dynamic-require */
-const createError = require('http-errors');
-const express = require('express');
-const path = require('path');
-const fs = require('fs');
-require('dotenv').config();
-const mongoose = require('mongoose');
-const pino = require('express-pino-logger')();
-const logger = require('./utils/logger');
+import createError from 'http-errors';
+import express from 'express';
+import path from 'path';
+import mongoose from 'mongoose';
+import dotenv from 'dotenv';
+import requireAll from 'require-all';
+import pino from 'express-pino-logger';
+import cors from 'cors';
+import logger from './utils/logger';
+// import authorize from './thirdparty/routes/authorize';
+// import slackRouter from './thirdparty/routes/slackRouter';
+// import IntegrationService from './controller/IntegrationService';
+import AuthenticationMiddleware from './middlewares/authentication';
+// import dataFetcher from './thirdparty/routes/DataFetcher';
 
-const app = express();
 const modelDirPath = path.join(__dirname, 'models');
+requireAll(modelDirPath);
+
+dotenv.config();
+const app = express();
+
 const routeDirPath = path.join(__dirname, 'routes');
+const publicFolder = path.join(__dirname, '../assets');
 
-app.use(pino);
+app.use(pino());
 app.use(express.json());
 app.use(express.urlencoded({ extended: false }));
-app.use(express.static(path.join(__dirname, 'public')));
+app.use(express.static(publicFolder));
+app.use('/api/v1', AuthenticationMiddleware);
+app.set('view engine', 'ejs');
```

Comment:
do you need a view engine?

Reply:
I thought of serving the OAuth popup page from server and I have to have scripts in the page for closing the popup window